### PR TITLE
[Do Not Merge][Draft] FORMS-5836 - Add Analytics support to CC Adapti…

### PR DESCRIPTION
**Pull request by sauraku**
…ve Forms

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

**Comment by rismehta** [comment]
Schema of the data layer needs to align to WCM, https://github.com/adobe/aem-core-wcm-components/blob/main/DATA_LAYER_INTEGRATION.md and additionally we should add event specific information

**Comment by rismehta** [comment]
You also need to make use of dataLayer attributes coming from server as part of sling model

**Comment by rismehta** [comment]
Please add test cases too

**Review by rismehta — CHANGES_REQUESTED** [review]
-1 for now

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/af-clientlibs/core-forms-components-runtime-all/js/datalayer.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+
+    const onDocumentReady = () => {
+        const dataLayerEnabled = document.body.hasAttribute("data-cmp-data-layer-enabled");
+        dataLayer = (dataLayerEnabled) ? window.adobeDataLayer = window.adobeDataLayer || [] : undefined;
+
+        if (dataLayerEnabled) {
+            registerGuideBridgeEvents();
+
+            window.addEventListener('beforeunload', function(){
```
Comment (arnehjain) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/af-clientlibs/core-forms-components-runtime-all/js/datalayer.js:111]: question: would this event be still triggered for successful form submissions? 

I think we might want to ensure that once a form has been submitted we don't mark it as abondened.

**Pull request closed without merging**